src/lightning/pytorch/trainer/trainer.py

```diff
@@ -965,33 +949,16 @@ def _teardown(self) -> None:
         self._signal_connector.teardown()
 
     def _run_stage(self) -> Optional[Union[_PREDICT_OUTPUT, _EVALUATE_OUTPUT]]:
-        # wait for all to join if on distributed
-        self.strategy.barrier("run-stage")
-
         if self.evaluating:
-            with self.profiler.profile(f"run_{self.state.stage}_evaluation"):
```

Comment:
This profiler call is gone, but the other top-level loop.run calls didn't have it. I can also add it to all if you think it's worth, or wrapping `_run_stage`